linter/lints/cpcps/lint_crl_has_idp.go

```diff
@@ -89,34 +89,32 @@ func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
 		}
 	}
 
-	if idpv.PeekASN1Tag(cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) || idpv.PeekASN1Tag(cryptobyte_asn1.Tag(1).ContextSpecific()) {
-		// If either the distributionPoint[0] field or the
-		// onlyContainsUserCerts[1] field is present, assume we're dealing with
+	distributionPointTag := cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()
+	onlyContainsUserCertsTag := cryptobyte_asn1.Tag(1).ContextSpecific()
+	onlyContainsCACertsTag := cryptobyte_asn1.Tag(2).ContextSpecific()
+	distributionPointURITag := cryptobyte_asn1.Tag(6).ContextSpecific()
+
+	if idpv.PeekASN1Tag(distributionPointTag) || idpv.PeekASN1Tag(onlyContainsUserCertsTag) {
+		// If either the distributionPoint [0] field or the
+		// onlyContainsUserCerts [1] field is present, assume we're dealing with
 		// a CRL containing Subscriber Certs.
 
-		// We read this boolean as a byte and ensure the value is 0xFF because
-		// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for
-		// the onlyContainsUserCerts[1] and onlyContainsCACerts[2] tagged
-		// fields. X.690 (07/2002) section 8.2 states that a boolean set to true
-		// will have contents FF.
-		// https://www.itu.int/rec/T-REC-X.690-200207-S/en
-		onlyContainsUserCerts := make([]byte, 0)
 		var dpName cryptobyte.String
-		if !idpv.ReadASN1(&dpName, cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) {
+		if !idpv.ReadASN1(&dpName, distributionPointTag) {
 			return &lint.LintResult{
 				Status:  lint.Warn,
 				Details: "Failed to read IDP distributionPoint",
 			}
 		}
-		if !dpName.ReadASN1(&dpName, cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) {
+		if !dpName.ReadASN1(&dpName, distributionPointTag) {
```

Comment:
The aliasing of the tags is really helpful here! For instance I notice that we're trying to parse the `distributionPointTag` twice from two different levels of nesting. Really this second one should be `distributionPointFullNameTag`, which happens to have the same value (`cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()`).